src/SourceLocator/Type/DirectoriesSourceLocator.php

```diff
@@ -55,11 +56,8 @@ public function locateIdentifier(Reflector $reflector, Identifier $identifier):
         return $this->aggregateSourceLocator->locateIdentifier($reflector, $identifier);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public function locateIdentifiersByType(Reflector $reflector, IdentifierType $identifierType): array
+    public function locateIdentifiersByType(Reflector $reflector, IdentifierType $identifierType): Generator
```

Comment:
The implementation of this class can be rolled back, except for the return type being `iterable`